packages/@aws-cdk/aws-s3-deployment/lib/lambda/index.py

```diff
@@ -154,12 +155,16 @@ def s3_deploy(s3_source_zips, s3_dest, user_metadata, system_metadata, prune, ex
             s3_source_zip = s3_source_zips[i]
             markers       = source_markers[i]
 
-            archive=os.path.join(workdir, str(uuid4()))
-            logger.info("archive: %s" % archive)
-            aws_command("s3", "cp", s3_source_zip, archive)
-            logger.info("| extracting archive to: %s\n" % contents_dir)
-            logger.info("| markers: %s" % markers)
-            extract_and_replace_markers(archive, contents_dir, markers)
+            if extract:
+                archive=os.path.join(workdir, str(uuid4()))
+                logger.info("archive: %s" % archive)
+                aws_command("s3", "cp", s3_source_zip, archive)
+                logger.info("| extracting archive to: %s\n" % contents_dir)
+                logger.info("| markers: %s" % markers)
+                extract_and_replace_markers(archive, contents_dir, markers)
```

Comment:
It looks like marker replacement won't work anymore if `extract == false`.

Check for the combination of markers and `extract == false` in the constructor of the `BucketDeployment` and throw an error that these features cannot be combined.

Reply:
That's correct. I used the existing `hasMarkers` const in a new conditional that checks if sources have markers and that extract is false, and then it throws an error on construction of `BucketDeployment`. I added a unit test for this as well.